File: nodejs/7 kyu/53.js
// https://www.codewars.com/kata/5a3dd29055519e23ec000074/train/javascript

// Passed

function checkExam(array1, array2) {
  const scores = array2.map((submitted_answer, idx) => {
    if (submitted_answer.trim().length === 0) {
      // Blank answer
      return 0;
    }

    return submitted_answer === array1[idx] ? 4 : -1;
  });

  let totalScore = Math.max(
    scores.reduce((prev, curr) => prev + curr, 0),
    0
  ); // prevents -ve score

  // Returning total score
  return totalScore;
}

const output = checkExam(["a", "a", "b", "b"], ["a", "c", "b", "d"]);
console.log(output);

File: python/7 kyu/53.py
# https://www.codewars.com/kata/5a090c4e697598d0b9000004/train/python

# Passed

def solve(arr):
    res = list()
    arr.sort()
    total_numbers = len(arr)
    for idx in range(total_numbers // 2):
        res.extend([arr[total_numbers - 1 - idx], arr[idx]])

    if total_numbers % 2:
        # we have odd number of total numbers
        res.append(arr[total_numbers//2])

    return res

output = solve([15,11,10,7,12])
print(output)